integrations/pjrt/third_party/pjrt_c_api/xla/xla_data.proto

```diff
@@ -0,0 +1,1153 @@
+  // Finite and NaN values are supported. Unlike IEEE types, infinities are not
+  // supported.  NaN is represented when the exponent and mantissa bits are all
+  // 1s. All other values are finite.
+  //
+  // F8E4M3B11FNUZ has 4 exponent bits and 3 mantissa bits and a bias of 11. The
+  // "FNUZ" means only Finite and NaN values are supported; zero is unsigned.
+  // Unlike IEEE types, infinities are not supported.  NaN is represented when
+  // the exponent and mantissa bits are all 0s with a sign bit of 1. All other
+  // values are finite.
+  //
+  // F8E3M4 has 3 exponent bits and 4 mantissa bits, and is similar to the
+  // existing IEEE types.
+  //
+  // Support for these dtypes is under development. They do not yet work
+  // properly in most cases.
+  // TODO(b/259609697): Fully support FP8.
+  F8E5M2 = 19;
+  F8E4M3 = 28;
+  F8E4M3FN = 20;
+  F8E4M3B11FNUZ = 23;
+  F8E3M4 = 29;
+
+  // FP8 dtypes, as described in this paper: https://arxiv.org/abs/2206.02915
+  //
+  // F8E5M2FNUZ has 5 exponent bits and 2 mantissa bits.
+  // F8E4M3FNUZ has 4 exponent bits and 3 mantissa bits.
+  //
+  // The "FNUZ" means only Finite and NaN values are supported; zero is
+  // unsigned. Unlike IEEE types, infinities are not supported.  NaN is
+  // represented when the exponent and mantissa bits are all 0s with a sign bit
+  // of 1. All other values are finite.
+  //
+  // These differences mean there's an additional exponent value available. To
+  // keep the same dynamic range as an IEEE-like FP8 type, the exponent is
+  // biased one more than would be expected given the number of exponent bits
+  // (8 for Float8E4M3FNUZ and 16 for Float8E5M2FNUZ).
+  F8E5M2FNUZ = 24;
+  F8E4M3FNUZ = 25;
+
+  // Complex values of fixed width.
+  C64 = 15;   // Paired F32 (real, imag), as in std::complex<float>.
+  C128 = 18;  // Paired F64 (real, imag), as in std::complex<double>.
+
+  // A tuple is a polymorphic sequence; e.g. a shape that holds different
+  // sub-shapes. They are used for things like returning multiple values from a
+  // computation; e.g. a computation that returns weights and biases may have a
+  // signature that results in a tuple like (f32[784x2000], f32[2000])
+  //
+  // If a shape proto has the tuple element type, it may not have any entries
+  // in the dimensions field.
+  TUPLE = 13;
+
+  // An opaque type used for passing context-specific data to a custom
+  // operation. Shapes of this primitive type will have empty dimensions and
+  // tuple_shapes fields.
+  //
+  // (OPAQUE would be a better name for this identifier, but that conflicts with
+  // a macro defined in windows.h.)
+  OPAQUE_TYPE = 14;
+
+  // A token type threaded between side-effecting operations. Shapes of this
+  // primitive type will have empty dimensions and tuple_shapes fields.
+  TOKEN = 17;
+
+  // Next = 30
+}
+// LINT.ThenChange(
+//   https://www.tensorflow.org/code/tensorflow/compiler/xla/tools/driver.cc
+// )
```

Comment:
Yeah there's many broken links in XLA since it's separated out from TensorFlow. The path `compiler/xla` is now moved to `third_party/xla` and is syncing with the `openxla/xla` repo.